sdk/sdk-view/src/test/java/mil/emp3/api/PointTest.java

```diff
@@ -0,0 +1,434 @@
+package mil.emp3.api;
+
+import android.test.mock.MockContext;
+import android.util.Log;
+import android.webkit.URLUtil;
+
+import org.bouncycastle.asn1.ASN1Sequence;
+import org.cmapi.primitives.GeoFillStyle;
+import org.cmapi.primitives.GeoIconStyle;
+import org.cmapi.primitives.GeoLabelStyle;
+import org.cmapi.primitives.GeoPoint;
+import org.cmapi.primitives.GeoPosition;
+import org.cmapi.primitives.GeoStrokeStyle;
+import org.cmapi.primitives.IGeoAltitudeMode;
+import org.cmapi.primitives.IGeoColor;
+import org.cmapi.primitives.IGeoFillStyle;
+import org.cmapi.primitives.IGeoIconStyle;
+import org.cmapi.primitives.IGeoLabelStyle;
+import org.cmapi.primitives.IGeoPoint;
+import org.cmapi.primitives.IGeoPosition;
+import org.cmapi.primitives.IGeoStrokeStyle;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.powermock.api.mockito.PowerMockito.when;
+import org.junit.*;
+
+import mil.emp3.api.enums.FeatureTypeEnum;
+import sec.geo.kml.KmlOptions;
+
+/**
+ * Created by Matt.Miller on 10/16/2017.
+ */
+@PrepareForTest({URLUtil.class})
+public class PointTest extends TestBase {
+
+    public static final String TAG = PointTest.class.getSimpleName();
+    private static final double Epsilon = 1e-8;
+    Point p1;
+    private GeoPoint gp;
+
+    @Before
+    public void setUp() throws Exception {
+        PowerMockito.mockStatic(URLUtil.class);
+        when(URLUtil.isValidUrl(any(String.class))).thenReturn(true);
+        p1 = new Point();
+    }
+
+    private void compareGeoPosition(final IGeoPosition p1, final IGeoPosition p2) {
+        if(p1 == null && p2 == null) {
+            return;
+        }
+        assertEquals(p1.getLatitude(), p2.getLatitude(), Epsilon);
+        assertEquals(p1.getLongitude(), p2.getLongitude(), Epsilon);
+        assertEquals(p1.getAltitude(), p2.getAltitude(), Epsilon);
+    }
+
+    private void compareGeoPositionArray(final List<IGeoPosition> p1, final List<IGeoPosition> p2) {
+        final double size = p1.size();
+        if(size != p2.size()){
+            fail();
+        }
+        for(int i = 0; i < size; i++) {
+            compareGeoPosition(p1.get(i), p2.get(i));
+        }
+    }
+
+    private void compareGeoColor(final IGeoColor c1, final IGeoColor c2) {
+        assertEquals(c1.getAlpha(), c2.getAlpha(), Epsilon);
```

Comment:
how come this one is the only one with the Epsilon value? is it the only double?

Reply:
The other ones refer to rgb values which have to be integers.